core/src/test/scala/org/apache/spark/util/logging/DriverLoggerSuite.scala

```diff
@@ -0,0 +1,116 @@
+import org.apache.spark.{SparkContext, SparkFunSuite}
+import org.apache.spark.internal.config._
+import org.apache.spark.launcher.SparkLauncher
+import org.apache.spark.network.util.JavaUtils
+import org.apache.spark.util.Utils
+
+class DriverLoggerSuite extends SparkFunSuite with LocalSparkContext {
+
+  private val DRIVER_LOG_DIR_DEFAULT = "/tmp/hdfs_logs"
+
+  override def beforeAll(): Unit = {
+    FileUtils.forceMkdir(FileUtils.getFile(DRIVER_LOG_DIR_DEFAULT))
+  }
+
+  override def afterAll(): Unit = {
+    JavaUtils.deleteRecursively(FileUtils.getFile(DRIVER_LOG_DIR_DEFAULT))
+  }
+
+  test("driver logs are persisted") {
+    val sc = getSparkContext()
+
+    val app_id = sc.applicationId
+    // Run a simple spark application
+    sc.parallelize(1 to 1000).count()
+
+    // Assert driver log file exists
+    val rootDir = Utils.getLocalDir(sc.getConf)
+    val driverLogsDir = FileUtils.getFile(rootDir, "driver_logs")
+    assert(driverLogsDir.exists())
+    val files = driverLogsDir.listFiles()
+    assert(files.length === 1)
+    assert(files(0).getName.equals("driver.log"))
+
+    sc.stop()
+    // On application end, file is moved to Hdfs (which is a local dir for this test)
+    assert(!driverLogsDir.exists())
+    val hdfsDir = FileUtils.getFile(sc.getConf.get(DRIVER_LOG_DFS_DIR).get, app_id)
+    assert(hdfsDir.exists())
+    val hdfsFiles = hdfsDir.listFiles()
+    assert(hdfsFiles.length > 0)
+    JavaUtils.deleteRecursively(hdfsDir)
+    assert(!hdfsDir.exists())
+  }
+
+  test("driver logs are synced to hdfs continuously") {
+    val sc = getSparkContext()
+
+    val app_id = sc.applicationId
+    // Run a simple spark application
+    sc.parallelize(1 to 1000).count()
+
+    // Assert driver log file exists
+    val rootDir = Utils.getLocalDir(sc.getConf)
+    val driverLogsDir = FileUtils.getFile(rootDir, "driver_logs")
+    assert(driverLogsDir.exists())
+    val files = driverLogsDir.listFiles()
+    assert(files.length === 1)
+    assert(files(0).getName.equals("driver.log"))
+    for (i <- 1 to 1000) {
+      logInfo("Log enough data to log file so that it can be flushed")
+    }
+
+    // Sync the driver logs manually instead of waiting for scheduler
+    sc._driverLogger.foreach(_.writer.foreach(_.run()))
+    val hdfsDir = FileUtils.getFile(sc.getConf.get(DRIVER_LOG_DFS_DIR).get, app_id)
+    assert(hdfsDir.exists())
+    val hdfsFiles = hdfsDir.listFiles()
+    assert(hdfsFiles.length > 0)
+    val driverLogFile = hdfsFiles.filter(f => f.getName.equals("driver.log")).head
+    val hdfsIS = new BufferedInputStream(new FileInputStream(driverLogFile))
+    assert(hdfsIS.available() > 0)
+
+    sc.stop()
+    // Ensure that the local file is deleted on application end
+    assert(!driverLogsDir.exists())
+    JavaUtils.deleteRecursively(hdfsDir)
+    assert(!hdfsDir.exists())
+  }
+
+  private def getSparkContext(): SparkContext = {
+    val conf = new SparkConf()
+    conf.set("spark.local.dir", "/tmp")
```

Comment:
Don't do this. Tests run with a pre-configured temp directory, and should not write to `/tmp`.

If you need to create a temp directory, do so explicitly using the existing `Utils` API.